### PR TITLE
feat: Added resource github_actions_default_workflow_repository_permissions

### PR DESCRIPTION
**Pull request by petergvizd-s1**
Resolves https://github.com/integrations/terraform-provider-github/issues/1266

----

## Behavior
Added resource for setting default workflow permissions for repository via API calls 
* https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#set-default-workflow-permissions-for-a-repository
* https://docs.github.com/en/rest/actions/permissions?apiVersion=2022-11-28#get-default-workflow-permissions-for-a-repository

### Before the change?

* Currently there is no resource for managing default workflow permissions for repository

### After the change?

* Added resource `github_actions_default_workflow_repository_permissions` for setting default workflow permissions for repository

### Other information

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

File: vendor/github.com/google/go-github/v53/github/github-accessors.go [hunk 1]
```diff
@@ -166,6 +166,22 @@ func (a *ActionsPermissionsRepository) GetEnabled() bool {
 	return *a.Enabled
 }
 
+// GetDefaultWorkflowPermissions returns the DefaultWorkflowPermissions field if it's non-nil, zero value otherwise.
```
Comment (kfcampbell) [inline vendor/github.com/google/go-github/v53/github/github-accessors.go:169]: Did you make these updates to vendored files manually? How did this come about without associated go.mod and go.sum files?

**Review by bitcoinfinancier — APPROVED** [review]

**Comment by kfcampbell** [comment]
@petergvizd-s1 I'm going to close this now since it's been sitting for a while; please reopen if you'd like to continue this work.

**Comment by saminahbab** [comment]
This should really be opened again, as it is not possible in my use case to use terraform to create repositories and create github actions workflows that can push branches in semantic release

**Comment by prores-raypet** [comment]
Why was this closed? It's exactly what we need for terraforming our projects properly.

**Comment by kfcampbell** [comment]
I closed the PR as it was abandoned. Please feel free to take the work and reopen a similar PR if you'd like to drive it forward!

**Pull request closed without merging**